plugins/modules/podman_image.py

```diff
@@ -781,19 +781,12 @@ def push_image(self):
         # Build the destination argument
         dest = self.push_args.get('dest')
         dest_format_string = '{dest}/{image_name}'
-        regexp = re.compile(r'/{name}(:{tag})?'.format(name=self.name, tag=self.tag))
         if not dest:
             if '/' not in self.name:
                 self.module.fail_json(msg="'push_args['dest']' is required when pushing images that do not have the remote registry in the image name")
 
```

Comment:
Doing so, we can do all the checks in one shot - and remove the `if not dest: fail` on L793-794
```suggestion
        if not dest:
            if transport:
                self.module.fail_json(msg="'ppush_args['transport'] requires 'push_args['dest'] but it was not provided.")
            if '/' not in self.name:
                self.module.fail_json(msg="'push_args['dest']' is required when pushing images that do not have the remote registry in the image name")
                
```
It's not much, but it  would allow to do all the "dest_string" building afterward, in a far more convenient way.
We could also apply a plain/stupid:
```YAML
if not dest:
  dest = self.name
```

so that we can stick to `dest` for all the rest of the code.

BTW: `transport` wouldn't work with a `name` matching a `dest` syntax?